src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -192,23 +230,80 @@ public void setNumberOfInstances(int numberOfInstances) {
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
-			}
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));				
+			}      
+			
+				v.setNumberOfInstances(sas.getInstances());
+				
+				return Mono.just(v);
+		});
 			
-			v.setNumberOfInstances(sas.getInstances());
+		Flux<List<RouteResource>> routeApplicationFlux = initialOSAVectorFlux.flatMapSequential(v -> this.getAppRoutes(v.getApplicationId()));
+		Flux<OSAVector> osaVectorApplicationRouteFlux = Flux.zip(osaVectorApplicationFlux, routeApplicationFlux).flatMap(tuple -> {
+			OSAVector v = tuple.getT1();
 			
+			List<RouteResource> routes = tuple.getT2();
+
+			if (routes.size() == 0) {
+				// NB: This drops the current target!
+				return Mono.empty();
+			}
+		
+			// multiple routes for the app can be returned, lets find the matching one.
+			RouteResource metricsRoute = routes.stream()
+				.filter(r -> v.getAccessURL().contains(r.getUrl()))
+				.findFirst()
+				.orElse(null);
+
+			if (metricsRoute == null) {
+				return Mono.just(v);
+			}
+
+			// from the matching route lets find the destination matching our app
+			// a route can have many different destinations
+			// once we get this we can use the routeID and the domainID
+			Destination dest = metricsRoute.getDestinations()
+				.stream()
+				.filter(d -> d.getApplication().getApplicationId().equals(v.getApplicationId()))
+				.findFirst()
+				.orElse(null);
+						
+			if (dest != null) {
+				v.setInternalPort(dest.getPort());
```

Comment:
BTW: After having a read a little about container-to-container communication in CF, I understand that this is very implicit here. You assume that the application gets another route which is set like this:

``` 
cf map-route myapp app.internal --hostname myapphost --port 9090 
```

I think the scenario we are having here needs some further documentation as well, such that users will be able to configure their application properly.

Reply:
Yes I think this would need to be documented and might be where I need to allow for a default port at the target level for cases where the route port is null.

For now I will leave this as a TODO so that it can be implemented.
I will add test cases for it too.

I think the right solution is to add a default port at the target level so it can fall back to something configured by the suer.